test/common.sh

```diff
@@ -0,0 +1,276 @@
+            function_exists pre-apply-taskrun-hook && pre-apply-taskrun-hook
+
+            # Make sure we have deleted the content, this is in case of rerun
+            # and namespace hasn't been cleaned up or there is some Cluster*
+            # stuff, which really should not be allowed.
+            ${KUBECTL_CMD} -n ${tns} delete -f ${TMPF} >/dev/null 2>/dev/null || true
+            ${KUBECTL_CMD} -n ${tns} create -f ${TMPF}
+        done
+
+        resource_to_wait_for["$testname/${version}"]="${tns}|$started"
+
+    done
+
+    # I would refactor this to a function but bash limitation is too great, really need a rewrite the sooner
+    # the uglness to pass a hashmap to a function https://stackoverflow.com/a/17557904/145125
+    local cnt=0
+    local all_status=''
+    local reason=''
+    local maxloop=60 # 10 minutes max
+
+    set +x
+    while true;do
+        # If we have timed out then show failures of what's remaining in
+        # resource_to_wait_for we assume only first one fails this
+        [[ ${cnt} == "${maxloop}" ]] && {
+            for testname in "${!resource_to_wait_for[@]}";do
+                target_ns=${resource_to_wait_for[$testname]}
+                show_failure "${testname}" "${target_ns}"
+            done
+        }
+        [[ -z ${resource_to_wait_for[*]} ]] && {
+            break
+        }
+
+        for testname in "${!resource_to_wait_for[@]}";do
+            target_ns=${resource_to_wait_for[$testname]%|*}
+            started=${resource_to_wait_for[$testname]#*|}
+            # sometimes we don't get all_status and reason in one go so
+            # wait until we get the reason and all_status for 5 iterations
+            for tektontype in pipelinerun taskrun;do
+                for _ in {1..10}; do
+                    all_status=$(${KUBECTL_CMD} get -n ${target_ns} ${tektontype} --output=jsonpath='{.items[*].status.conditions[*].status}')
+                    reason=$(${KUBECTL_CMD} get -n ${target_ns} ${tektontype} --output=jsonpath='{.items[*].status.conditions[*].reason}')
+                    result_values=$(${KUBECTL_CMD} get -n ${target_ns} ${tektontype} --output=jsonpath='{.items[*].status.results[*].value}')
+                    [[ ! -z ${all_status} ]] && [[ ! -z ${reason} ]] && break
+                    sleep 1
+                done
+                # No need to check taskrun if pipelinerun has been set
+                [[ ! -z ${all_status} ]] && [[ ! -z ${reason} ]] && break
+                
+            done
+
+            if [[ -z ${all_status} || -z ${reason} ]];then
+                echo "Could not find a created taskrun or pipelinerun in ${target_ns}"
+            fi
+
+            breakit=True
+            for status in ${all_status};do
+                [[ ${status} == *ERROR || ${reason} == *Fail* || ${reason} == Couldnt* ]] && show_failure ${testname} ${target_ns}
+
+                if [[ ${status} != True ]];then
+                    breakit=
```

Comment:
Should we rather have false here?

Reply:
don't see any issue doing that. updated.